### PR TITLE
TEAMNADO-3272 Add create activation keys form

### PR DESCRIPTION
**Pull request by seodma**
This PR add Create Activation Key Form and Modal

### CODE REVIEW

File: src/Components/Forms/CreateActivationKeyForm.js
Comment (jmbuscemi) [inline src/Components/Forms/CreateActivationKeyForm.js]: Typo in "hyphens"

But I'm also questioning this entire message, so I want to confirm the content with Shane once he's back from PTO.  For example, do we want to actually enforce that it starts with a letter and only lowercase letters can be used? Because the backend has no such validation requirements.

File: src/Components/Forms/CreateActivationKeyForm.js
Comment (jmbuscemi) [inline src/Components/Forms/CreateActivationKeyForm.js]: missing the (SLA) in parentheses
"Service Level Agreement (SLA) determines ..."

**Review by jmbuscemi — COMMENTED** [review]
This looks great. Just a couple small typos and the "Loading" component we talked about in Slack when the create form is submitted.

File: src/Components/ActivationKeys/ActivationKeys.js
Comment (jmbuscemi) [inline src/Components/ActivationKeys/ActivationKeys.js]: This should have lower case "activation key"

File: src/Components/Modals/CreateActivationKeyModal.js
Comment (jmbuscemi) [inline src/Components/Modals/CreateActivationKeyModal.js]: Title should be "Create new activation key"

**Comment by jmbuscemi** [comment]
Apologies for submitting multiple review comments and not doing them all at once.

I found one more thing - the Acceptance Criteria mentions "Create button (disabled until valid name is entered)" and currently the button is enabled the whole time. You can't submit the form until the name is valid, but the button is still enabled.

**Comment by jmbuscemi** [comment]
From the perspective of the UI requirements of the story this looks good and meets requirements.  I did not review the actual React code (because of my lack of experience with React). Once that has been reviewed, I'm good with merging this.

File: src/hooks/__tests__/useCreateActivationKey.test.js
Comment (bastilian) [inline src/hooks/__tests__/useCreateActivationKey.test.js:33]: This looks a bit odd. It might not be needed to await the act or pass it an async function, or is it?

**Review by bastilian — APPROVED** [review]
LGTM code wise! Only a minor thing in the tests, but they still seem to pass. :) Thank you @seodma!

File: src/hooks/__tests__/useCreateActivationKey.test.js
Comment (seodma) [reply src/hooks/__tests__/useCreateActivationKey.test.js:33]: yes, In this case I need to wait for async callback, may be it's not a best way to test it, I am always open to suggestions.

**Pull request merged**
